src/npm-dependency-version-checker.js

```diff
@@ -32,20 +33,28 @@ class NPMDependencyVersionChecker extends DependencyVersionChecker {
     let basedir = addon.root || getProject(addon).root;
 
     let jsonPath;
+    let key = target + 0x00 + basedir;
 
-    try {
-      // the custom `pnp` code here can be removed when yarn 1.13 is the
-      // current release this is due to Yarn 1.13 and resolve interoperating
-      // together seemlessly
-      jsonPath = pnp
-        ? pnp.resolveToUnqualified(target, basedir)
-        : resolve.sync(target, { basedir });
-    } catch (e) {
-      if (ALLOWED_ERROR_CODES.includes(e.code)) {
-        jsonPath = null;
-      } else {
-        throw e;
+    if (cache.has(key)) {
+      jsonPath = cache.get(key);
+    } else {
+
+      try {
+        // the custom `pnp` code here can be removed when yarn 1.13 is the
+        // current release this is due to Yarn 1.13 and resolve interoperating
+        // together seemlessly
+        jsonPath = pnp
+          ? pnp.resolveToUnqualified(target, basedir)
+          : resolve.sync(target, { basedir });
```

Comment:
Do you want to do something like is done in hash-for-dep here, instead of resolve.sync(). That's the really expensive one.

Reply:
Ya, i'm going to make fast-require-resolve that supports both require.resolve and pnp and caches tonight. I'll steal code from your hash-for-dep code to do so.

Reply:
cool - thanks!